arch/arc/core/swap.S

```diff
@@ -92,7 +92,11 @@ SECTION_FUNC(TEXT, __swap)
 #endif
 
 #ifdef CONFIG_ARC_HAS_SECURE
+#ifdef CONFIG_ARC_SECURE_FIRMWARE
 	lr r3, [_ARC_V2_SEC_STAT]
+#else
+	mov r3, 0
```

Comment:
See earlier remark on ERSEC_STAT, I think it's ok to access from Normal mode as well.